src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {OwnableOperable} from "./OwnableOperable.sol";
+import {IERC20} from "./Interfaces.sol";
+
+abstract contract AbstractARM is OwnableOperable {
+    /// @notice The swap input token that is transferred to this contract.
+    /// From a User perspective, this is the token being sold.
+    /// token0 is also compatible with the Uniswap V2 Router interface.
+    IERC20 public immutable token0;
+    /// @notice The swap output token that is transferred from this contract.
+    /// From a User perspective, this is the token being bought.
+    /// token1 is also compatible with the Uniswap V2 Router interface.
+    IERC20 public immutable token1;
+
+    uint256[50] private _gap;
```

Comment:
Gap adjust. :D

Reply:
`token0` and `token1` are immutable so don't take a storage slot.

Reply:
My bad